source/shared/core_conn.cpp

```diff
@@ -233,17 +333,84 @@ sqlsrv_conn* core_sqlsrv_connect( _In_ sqlsrv_context& henv_cp, _In_ sqlsrv_cont
     return return_conn;
 }
 
+// core_compare_error_state
+// This method compares the error state to the one specified
+// Parameters:
+// conn         - the connection structure on which we establish the connection 
+// rc           - ODBC return code
+// Return       - a boolean flag that indicates if the error states are the same
+
+bool core_compare_error_state( _In_ sqlsrv_conn* conn,  _In_ SQLRETURN rc, _In_ const char* error_state )
+{
+    if( SQL_SUCCEEDED( rc ) ) 
+        return false;
+
+    SQLCHAR state[ SQL_SQLSTATE_BUFSIZE ];
+    SQLSMALLINT len;
+    SQLRETURN sr = SQLGetDiagField( SQL_HANDLE_DBC, conn->handle(), 1, SQL_DIAG_SQLSTATE, state, SQL_SQLSTATE_BUFSIZE, &len );
+
+    bool same = false;
+    if( SQL_SUCCEEDED(sr) && ! strcmp(error_state, reinterpret_cast<char*>( state ) ) )
+        same = true;
+
+    return same;
+}
+
+// core_search_odbc_driver_unix
+// This method is meant to be used in a non-Windows environment, 
+// searching for a particular ODBC driver name in the odbcinst.ini file
+// Parameters:
+// driver_version   - a valid value in enum DRIVER_VERSION
+// Return           - a boolean flag that indicates if the specified driver version is found or not
+
+bool core_search_odbc_driver_unix( _In_ DRIVER_VERSION driver_version )
+{
+    char szBuf[DEFAULT_CONN_STR_LEN+1];     // use an arbitrary large enough buffer size
+    WORD cbBufMax = DEFAULT_CONN_STR_LEN; 
+    WORD cbBufOut;
+    char *pszBuf = szBuf;
+    bool found_driver = false;
+
+#ifndef _WIN32
+    // get all the names of the installed drivers delimited by null characters
+    if(! SQLGetInstalledDrivers( szBuf, cbBufMax, &cbBufOut ) )
+    {
+        return false;
+    }
+
+    // extract the ODBC driver name 
+    std::string driver = CONNECTION_STRING_DRIVER_NAME[ DRIVER_VERSION(driver_version) ];
+    std::size_t pos1 = driver.find_first_of("{");
+    std::size_t pos2 = driver.find_first_of("}");
+    std::string driver_str = driver.substr( pos1 + 1, pos2 - pos1 - 1);
+
+    // search for the ODBC driver...
+    const char* driver_name = driver_str.c_str();
+    do
+    {
+        if( strstr( pszBuf, driver_name ) != 0  )
+        {
+            found_driver = true;
+            break;
```

Comment:
do while (blah && !found_driver)

Reply:
I like the second suggestion better because I don't want it to proceed to get the next driver